dashboard/app/controllers/api/v1/sections_controller.rb

```
@@ -12,4 +12,50 @@ def index
   def show
     render json: @section.summarize
   end
+
+  # POST /api/v1/sections
+  # Create a new section
+  def create
+    # Validate login_type
+    # TODO: Push validation into model?
+    login_type = (['none', '', nil].include? params[:login_type]) ? 'email' : params[:login_type]
+    return head :bad_request unless %w(word picture email).include? login_type
+
+    valid_script = params[:script] && valid_script_id?(params[:script][:id])
+    script_to_assign = valid_script && Script.find(params[:script][:id])
+
+    section = Section.create!(
```

Might be better to do the non-bang create, and branch on `section.valid?` so we can `render head :bad_request` when it fails instead of blowing up.

---

Remind me - does the non-bang `create` mean I have to call `save!` later? E.g.
```ruby
section = Section.create(...)
if section.valid?
  section.save
  render json: section.summarize
else
  head :bad_request
end
```

---

No need. [create](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/persistence.rb#L33-L34) is basically short for `new` and `save`. The only difference (in general) between the bang methods and the non-bang ones is the `!` causes it to raise an error on failure instead of returning boolean.